lib/cache/sqlite-cache-store.js

```diff
@@ -0,0 +1,380 @@
+     */
+    const result = {
+      body: value.body ? parseBufferArray(JSON.parse(value.body)) : null,
+      statusCode: value.statusCode,
+      statusMessage: value.statusMessage,
+      rawHeaders: value.rawHeaders ? parseBufferArray(JSON.parse(value.rawHeaders)) : undefined,
+      etag: value.etag ? value.etag : undefined,
+      cachedAt: value.cachedAt,
+      staleAt: value.staleAt,
+      deleteAt: value.deleteAt
+    }
+
+    return result
+  }
+
+  /**
+   * @param {import('../../types/cache-interceptor.d.ts').default.CacheKey} key
+   * @param {import('../../types/cache-interceptor.d.ts').default.CacheValue} value
+   * @returns {Writable | undefined}
+   */
+  createWriteStream (key, value) {
+    assertCacheKey(key)
+    assertCacheValue(value)
+
+    const url = this.#makeValueUrl(key)
+    let currentSize = 0
+    /**
+     * @type {Buffer[] | null}
+     */
+    let body = key.method !== 'HEAD' ? [] : null
+    const maxEntrySize = this.#maxEntrySize
+    const findValue = this.#findValue.bind(this)
+    const updateValueQuery = this.#updateValueQuery
+    const insertValueQuery = this.#insertValueQuery
+
+    const writable = new Writable({
+      write (chunk, encoding, callback) {
+        if (typeof chunk === 'string') {
+          chunk = Buffer.from(chunk, encoding)
+        }
+
+        currentSize += chunk.byteLength
+
+        if (body) {
+          if (currentSize >= maxEntrySize) {
+            body = null
+            this.end()
+            return callback()
+          }
+
+          body.push(chunk)
+        }
+
+        callback()
+      },
+      final (callback) {
+        if (body === null) {
+          return callback()
+        }
+
+        /**
+         * @type {SqliteStoreValue | undefined}
+         */
+        const existingValue = findValue(key, true)
+        if (existingValue) {
+          // Updating an existing response, let's delete it
+          updateValueQuery.run(
+            JSON.stringify(stringifyBufferArray(body)),
+            value.deleteAt,
+            value.statusCode,
+            value.statusMessage,
+            value.rawHeaders ? JSON.stringify(stringifyBufferArray(value.rawHeaders)) : null,
+            value.etag,
+            value.cachedAt,
+            value.staleAt,
+            value.deleteAt,
+            existingValue.id
+          )
+        } else {
+          // New response, let's insert it
+          insertValueQuery.run(
+            url,
+            key.method,
+            JSON.stringify(stringifyBufferArray(body)),
+            value.deleteAt,
+            value.statusCode,
+            value.statusMessage,
+            value.rawHeaders ? JSON.stringify(stringifyBufferArray(value.rawHeaders)) : null,
+            value.etag ? value.etag : null,
+            value.vary ? JSON.stringify(value.vary) : null,
+            value.cachedAt,
+            value.staleAt,
+            value.deleteAt
+          )
+        }
```

Comment:
What about https://www.sqlite.org/lang_upsert.html?

Reply:
I'm not sure how we could incorporate that since the only unique key we have is the id